### ISSUE TITLE
MonomorphismsDigraphs returns non-monomorphisms

### ISSUE THREAD

**Issue by wilfwilson** [gap-packages/Digraphs#222]
The following was reported by Gordon Royle to James Mitchell via email:
```
gap> D1 := DigraphFromGraph6String("E}hO");
<digraph with 6 vertices, 18 edges>
gap> D2 := DigraphFromGraph6String("E}h_");
<digraph with 6 vertices, 18 edges>
gap> mono := MonomorphismsDigraphs(D1, D2);
[ Transformation( [ 1, 2, 3, 3 ] ), Transformation( [ 1, 3, 2, 2, 4 ] ),
  Transformation( [ 2, 1, 3, 3, 6, 5 ] ),
  Transformation( [ 2, 3, 1, 1, 4, 5 ] ),
  Transformation( [ 3, 1, 2, 2, 6, 4 ] ),
  Transformation( [ 3, 2, 1, 1, 5, 4 ] ) ]
gap> ForAll(mono, f -> IsDigraphHomomorphism(D1, D2, f));
true
gap> ForAny(mono, f -> IsDigraphMonomorphism(D1, D2, f));
false
```
`MonomorphismsDigraphs` is returning transformations that, although homomorphisms, do not define monomorphisms. (They're clearly not injective!).

**Comment by james-d-mitchell** [comment]
Thanks @wilfwilson I believe I have a fix for this, I’ll make Pr tomorrow

**Comment by wilfwilson** [comment]
Great news.

**Comment by wilfwilson** [comment]
Resolved in v0.15.4.
